src/actions/create.ts

```diff
@@ -42,32 +31,41 @@ export const createEvent = async function createEvent({
   name: string;
   owner: string;
 }): Promise<EVENT_RESULT> {
-  const player = await getPlayer({ guildId, discordId: owner });
+  const playerRepository = getRepository(Player);
+  const eventRepository = getRepository(Event);
+  const player = await playerRepository.findOne({
+    where: { discordId: owner },
+  });
+
+  if (!player) {
+    throw new Error(
+      `Player not found during create event process ID: ${owner}`
+    );
+  }
 
   // Check if player already owns an event
-  let event = await getEvent({ guildId, owner: player.id });
-  if (event) {
+  const eventExists = await eventRepository.count({
+    where: { guildId, owner: player },
+  });
+  if (eventExists) {
     // Limiting players to a single event
     console.log('Player already owns an event');
     return EVENT_RESULT.ALREADY_OWNED;
   }
 
-  const eventCount = await dbget<{ count: number }>({
-    sql: eventCountSql,
-    params: [guildId],
-  });
-  const newEventNumber = eventCount.count as keyof typeof numberToEmoji;
+  const guildEventCount = await eventRepository.count({ where: guildId });
+  const newEventNumber = guildEventCount as keyof typeof numberToEmoji;
 
   if (newEventNumber <= 10) {
     const emoji = numberToEmoji[newEventNumber];
-    await run({
-      sql: createEventSql,
-      params: [guildId, name, player.id, emoji],
-    });
-
-    event = await getEvent({ guildId, owner: player.id });
-    // Add owner to event
-    await run({ sql: addOwnerSql, params: [event.id, player.id] });
+    const newEvent = new Event();
+    newEvent.guildId = guildId;
+    newEvent.name = name;
+    newEvent.emoji = emoji;
+    newEvent.rank = 'iron1';
+    newEvent.owner = player;
+    newEvent.players = [player];
+    await eventRepository.save(newEvent);
```

Comment:
Could this not be reduced to something like:

```
const newEvent new Event({
 guildId: guildid,
 name: name
})
```
? Or just personal preference

Reply:
Their documentation had it this way but let me see if I can make constructors. We would still of course need to do it this way for modifying existing objects but for new ones we can probably use a constructor. Ill try it.

Reply:
Okay. Added constructors!